### ISSUE TITLE
Replace request.SeekableFileObject with a mmap-ed buffer

### ISSUE THREAD

**Issue by FirefoxMetzger** [imageio/imageio#616]
I've noticed that for some of the input types we perform manual buffering via (`SeekableFileObject`). One problem that I can see in the current implementation is that there the buffer is purely kept in memory, which may be undesirable behavior for *very* large files.

I was wondering if buffering inside a memory-mapped container - based on `mmap` and using a `tmpfile` -  could be a more efficient solution, or if it was explored and discarded previously.

It could also be a nice opportunity to expose our codebase to mmap, which promises some speed increase for I/O due to fewer system calls (potentially) and one less copy of the data..

**Comment by almarklein** [comment]
No, this has not been attempted or explored. Sounds like something that could work!
